lib/Mojo/WebSocketProxy/Dispatcher/Parser.pm

```diff
@@ -0,0 +1,104 @@
+package Mojo::WebSocketProxy::Dispatcher::Parser;
+
+use strict;
+use warnings;
+
+use Mojo::WebSocketProxy::Dispatcher::Config;
+
+sub parse_req {
+    my ($c, $req_storage) = @_;
+
+    my $result;
+    my $args = $req_storage->{args};
+    if (ref $args ne 'HASH') {
+        # for invalid call, eg: not json
+        $req_storage->{args} = {};
+        $result = $c->wsp_error('error', 'BadRequest', 'The application sent an invalid request.');
+    }
+
+    $result = _check_sanity($c, $req_storage) unless $result;
+
+    return $result;
+}
+
+sub _check_sanity {
```

Comment:
Maybe this comment helps https://github.com/binary-com/perl-Mojo-WebSocketProxy/pull/1/files#diff-e010619b1d5454359570e5ed3a0d624bR69